### PR TITLE
Extract database initialisation

### PR DESCRIPTION
**Pull request by seallard**
## Description
This PR extracts the database initialisation to a separate module `database.py`, which is the common and recommended way of doing it, instead of keeping it in the `Store` class. See [sqlalchemy docs](https://docs.sqlalchemy.org/en/20/orm/session_basics.html#when-do-i-make-a-sessionmaker).

### Fixed
- Extract db initialisation to separate module

This [version](https://semver.org/) is a:
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

**Review by ChrOertlin — APPROVED** [review]
Looks good! I suppose the same comments as outlined here: https://github.com/Clinical-Genomics/cg/pull/2584

File: housekeeper/cli/core.py [hunk 1]
```diff
@@ -44,7 +45,8 @@ def base(
         raise click.Abort
     context.obj["database"] = db_path
     LOG.info("Use root path %s", root_path)
-    context.obj["store"] = Store(db_path, root_path)
+    initialise_database(db_path)
+    context.obj["store"] = Store(root=root_path)
```
Comment (seallard) [inline housekeeper/cli/core.py:49]: Given that the initialisation no longer occurs in the `Store` class constructor but instead here in the base CLI command, will it be configured correctly when used in `cg`?

Comment (seallard) [reply housekeeper/cli/core.py:49]: Nope, this is how it is initialised in `cg`:
```
    def __init__(self, config: dict) -> None:
        self._store = Store(config["housekeeper"]["database"], config["housekeeper"]["root"])
        self.root_dir: str = config["housekeeper"]["root"]
```

File: housekeeper/cli/init.py [hunk 2]
```diff
@@ -14,18 +14,16 @@
 @click.pass_context
 def init(context, reset, force):
     """Setup the database."""
-    store: Store = context.obj["store"]
-    inspector = inspect(store.engine)
-    existing_tables = inspector.get_table_names()
+    existing_tables = get_tables()
```
Comment (henrikstranneheim) [inline housekeeper/cli/init.py:17]: Add type hint

File: housekeeper/store/database.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+
+SESSION_FACTORY: Optional[Session] = None
+ENGINE = None
+
+
+def initialise_database(db_uri: str):
+    """Initialize the global engine and session factory for SQLAlchemy."""
+    global SESSION_FACTORY, ENGINE
+    ENGINE = create_engine(db_uri, pool_pre_ping=True)
```
Comment (henrikstranneheim) [inline housekeeper/store/database.py:16]: type hints

**Review by henrikstranneheim — APPROVED** [review]
👍

File: housekeeper/store/database.py (hunk 3 above)
Comment (seallard) [reply housekeeper/store/database.py:18]: Typed at the top of the file now where it is "declared"

File: housekeeper/cli/core.py (hunk 1 above)
Comment (seallard) [reply housekeeper/cli/core.py:49]: https://github.com/Clinical-Genomics/cg/pull/2612

**Pull request merged**